modules/ding_facetbrowser/ding_facetbrowser.module

```diff
@@ -90,14 +91,152 @@ function ding_facetbrowser_ctools_plugin_api($module, $api) {
 }
 
 /**
- * Create a custom form with checkbox elements for facets and ajax callback
+ * Get the year_span element.
+ *
+ * @param string $search
+ *   The searchkey.
+ * @param array $conditions
+ *   List of conditions.
+ *
+ * @return array
+ *   A year span form element container.
+ */
+function _ding_facetbrowser_form_year_span_element($search, array $conditions) {
+  // To get the correct counter for the span, we need to run the
+  // search without the actual date filter on, so the complete dataset
+  // is included.
+  $run_search = FALSE;
+  $count_search = array(
+    'conditions' => $conditions,
+    'searchkey' => $search,
+  );
+  $range = array();
+
+  // Remove all facet.date conditions.
+  if (!empty($count_search['conditions']['facets'])) {
+    foreach ($count_search['conditions']['facets'] as $delta => $condition_facet) {
+      if (strpos($condition_facet, 'facet.date:') === 0) {
+        $parts = explode(':', $condition_facet);
+
+        // Set the range to match the year span.
+        if (isset($parts[2])) {
+          switch ($parts[2]) {
+            case '>=';
+              $range[0] = (int) $parts[1];
+              break;
+
+            case '<=';
+              $range[1] = (int) $parts[1];
+              break;
+          }
+        }
+
+        $run_search = TRUE;
+        unset($count_search['conditions']['facets'][$delta]);
+      }
+    }
+
+    // Make sure the key values of the facets are re-numbered, so the hash will
+    // match later on.
+    $count_search['conditions']['facets'] = array_values($count_search['conditions']['facets']);
+  }
+
+  if (empty($count_search['conditions']['facets'])) {
+    unset($count_search['conditions']['facets']);
+  }
+
+  // Only run the counter, if it's different from the search already
+  // performed.
+  if ($run_search === TRUE) {
+    $cache_key = 'ding_facetbrowser:yearspan:' . md5(serialize($count_search));
+    $cache = cache_get($cache_key);
+
+    if (!$cache) {
+      $count_search['conditions']['result_id'] = 'ding_facetbrowser_count_search';
+      search_data($count_search['searchkey'], 'ting_search', $count_search['conditions']);
```

Comment:
Before you read this then please read my comment about `$search_result->conditions` further down:

If we have the search query and the filters available at this point then we can execute the query object directy without using the Drupal wrapper `search_data`. This provides us with greater control of the query which has a number of important benefits:

1. We can reduce the query to not include results (`$query->withCount(0)`) and only include the year facet so we do not spend time and resources retrieving anything we do not need.
2. We can increase the number of terms per facet to a very high number. The default set by `ting_search_search_execute` is 25 and when we only include actual facet values returned in the result in the slider this becomes important if we want to show the entire range.
3. We bypass the `ting_search_current_results()` static cache and can thus the changes in this PR which has do to with introducing result ids.